### DISCUSSION TITLE
Access useMutation isLoading state from another component

### DISCUSSION THREAD

**Discussion by jim-alexander** [TanStack/query#2707] [Q&A] [answered]
I want to be able to access the `isLoading` prop for a useMutation by multiple components but it doesn't seem to work like useQuery for example.

The scenario I have is that I want to click a button in a dialog to trigger a mutation where it will do a post request and close the dialog.

And I want to show that mutation loading state in other places.

I'm sure I have just missed some core concept or clientQuery usage but I can't seem to find anything similar.
I have also read that prop drilling is discouraged.

**Comment by jim-alexander** [comment]
After much research, I have found similar threads:
https://github.com/tannerlinsley/react-query/issues/2304
https://github.com/tannerlinsley/react-query/discussions/2294

**Comment by TkDodo** [comment, marked as answer]
Yes, there is an open issue. For your use case, you can use `useIsMutating`

**Reply by jim-alexander** [reply]
This does the job! Thank you for your prompt response.
